src/main/java/io/aiven/kafka/connect/transforms/ConcatFields.java

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.transforms;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ConcatFields<R extends ConnectRecord<R>> implements Transformation<R> {
+    private ConcatFieldsConfig config;
+    private static final Logger log = LoggerFactory.getLogger(ConcatFields.class);
+
+    @Override
+    public ConfigDef config() {
+        return ConcatFieldsConfig.config();
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        this.config = new ConcatFieldsConfig(configs);
+    }
+
+    @Override
+    public R apply(final R record) {
+        if (record.value() == null) {
+            throw new DataException("Value can't be null: " + record);
+        }
+        final R newRecord;
+        final List<Object> fieldValues = new ArrayList<>();
+
+        if (record.value() instanceof Struct) {
+            final Struct struct = (Struct) record.value();
+            final SchemaBuilder newSchema = SchemaBuilder.struct();
+            if (record.valueSchema() != null) {
+                record.valueSchema().fields().forEach(field -> newSchema.field(field.name(), field.schema()));
+            } else {
+                struct.schema().fields().forEach(field -> newSchema.field(field.name(), field.schema()));
+            }
+            newSchema.field(config.outputFieldName(), Schema.OPTIONAL_STRING_SCHEMA);
+            final Struct newStruct = new Struct(newSchema);
+            struct.schema().fields().forEach(field -> {
+                newStruct.put(field.name(), struct.get(field));
+            });
+            config.fieldNames().forEach(field -> {
+                try {
+                    if (struct.get(field) == null) {
+                        fieldValues.add(config.fieldReplaceMissing());
+                    } else {
+                        fieldValues.add(struct.get(field));
+                    }
+                } catch (final Exception e) {
```

Comment:
It'd be nice to not catch so broad an exception class and limit it to `DataException`, which is thrown when a filed is not found.